api/spec/requests/spree/api/coupon_codes_controller_spec.rb

```diff
@@ -12,6 +18,10 @@ module Spree
       let(:promo) { create(:promotion_with_item_adjustment, code: 'night_melody') }
       let(:promo_code) { promo.codes.first }
 
+      before do
+        allow_any_instance_of(Order).to receive_messages user: current_api_user
```

Comment:
Does the test fail without this line? Also, maybe we can assign the user to the order in the factory instead, WDYT?

Reply:
Yes, it fails **unless** we assign the user to the order, as you proposed. Both options are completely valid.